### PR TITLE
ramips: add support for Linksys RE7000

### PR DESCRIPTION
**Pull request by achterin**
Hardware specification:

- SoC: MediaTek MT7621AT (880 MHz)
- Flash: 16 MB (Macronix MX25L12835FM2I-10G)
- RAM: 128 MB (Nanya NT5CC64M16GP-DI)
- WLAN 2.4 GHz: 2x2 MediaTek MT7603EN
- WLAN 5 GHz: 2x2 MediaTek MT7615N
- Ethernet: 1x 10/100/1000 Mbps
- LED: Power, Wifi, WPS
- Button: Reset, WPS
- UART: 1:VCC, 2:GND, 3:TX, 4:RX (from LAN port)
- Serial console @ 57600,8n1

Flash instructions:

Connect to serial console and start up the device. As the bootloader got locked you need to type in a password to unlock U-Boot access. When you see the following output on the console:

relocate_code Pointer at: 87f1c000

type in the super secure password:

1234567890

Then select TFTP boot from RAM by selecting option 1 in the boot menu. As Linksys decided to leave out a basic TFTP configuration you need to set server- & client ip as well as the image filename the device will search for. You need to use the initramfs openwrt image for the TFTP boot process.

Once openwrt has booted up, upload the sysupgrade image via scp and run sysupgrade as normal.

### CODE REVIEW

**Comment by achterin** [comment]
can someone please review this? thanks in advance!

File: target/linux/ramips/dts/mt7621_linksys_re7000.dts [hunk 1]
```diff
@@ -0,0 +1,153 @@
+#include "mt7621.dtsi"
```
Comment (hauke) [inline target/linux/ramips/dts/mt7621_linksys_re7000.dts:1]: Please add a license header like:
`// SPDX-License-Identifier: GPL-2.0-or-later OR MIT`

File: target/linux/ramips/image/mt7621.mk [hunk 2]
```diff
@@ -1475,6 +1475,16 @@ define Device/linksys_re6500
 endef
 TARGET_DEVICES += linksys_re6500
 
+define Device/linksys_re7000
+  $(Device/uimage-lzma-loader)
+  IMAGE_SIZE := 16064k
+  DEVICE_VENDOR := Linksys
+  DEVICE_MODEL := RE7000
+  DEVICE_PACKAGES := kmod-mt7603 kmod-mt7615-firmware
+  SUPPORTED_DEVICES += re7000
```
Comment (hauke) [inline target/linux/ramips/image/mt7621.mk:1484]: `SUPPORTED_DEVICES` is only needed if you want to use sysupgrade from an older OpenWrt installation where the device was named `re7000`. I think there is no such installation, then you can remove this line.

File: target/linux/ramips/dts/mt7621_linksys_re7000.dts [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x30000>;
+				read-only;
+			};
+
+			partition@30000 {
+				label = "u-boot-env";
```
Comment (hauke) [inline target/linux/ramips/dts/mt7621_linksys_re7000.dts:74]: Please allow reading the U-Boot env by adding the configuration to package/boot/uboot-envtools/files/ramips.

**Comment by achterin** [comment]
@hauke Sorry for the delay. Fixed all your review comments.

**Pull request merged**